### PR TITLE
util/ck_errf.pl: add detection of unknown libcrypto and libssl libs

### PR DESCRIPTION
**Pull request by levitte**
The list of known libs are readily available in crypto/err/openssl.ec,
so lets use it to figure out if all error function codes belong to
known libs.

### CODE REVIEW

**Comment by levitte** [comment]
This is an alternative for #6452

File: util/ck_errf.pl [hunk 1]
```diff
@@ -19,6 +19,16 @@
 my $err_strict = 0;
 my $bad        = 0;
 
+my $config     = "crypto/err/openssl.ec";
```
Comment (richsalz) [inline util/ck_errf.pl]: I'd like a comment here explaining what's being done and why.

**Review by richsalz — APPROVED** [review]
Assuming you add the comment: approved.

**Comment by levitte** [comment]
Heh, you *know* that if you approve with a change request like that, I *will* ask you to reconfirm...

Comment looking good?

**Comment by richsalz** [comment]
comment is nice. reconfirm.

**Comment by levitte** [comment]
Merged.

a21180b70f6372fee836557df187d72f7a91b686 util/ck_errf.pl: add detection of unknown libcrypto and libssl libs

**Comment by DDvO** [comment]
I like the solution. 

Just two minor drawbacks (compared with the initial solution, which had other drawbacks):
* The error message just says 'unknown' - would be nice if it also said: 'in crypto/err/openssl.ec'. 
* In case there is no error in at the LIBerr() invocations but in oenssl.ec,  one error line would be sufficient.
  On the other hand I see the advantage of the new solution that it can also spot issues with the individual invocations.

Take for instance the ouput of `perl util/ck_errf.pl -strict */*.c */*/*.c`, which currently is:
```
engines/e_afalg.c:165:AFALG unknown
engines/e_afalg.c:189:AFALG unknown
engines/e_afalg.c:207:AFALG unknown
engines/e_afalg.c:356:AFALG unknown
engines/e_afalg.c:380:AFALG unknown
engines/e_afalg.c:387:AFALG unknown
engines/e_afalg.c:394:AFALG unknown
engines/e_afalg.c:722:AFALG unknown
engines/e_afalg.c:733:AFALG unknown
engines/e_afalg.c:739:AFALG unknown
engines/e_afalg.c:775:AFALG unknown
engines/e_afalg.c:792:AFALG unknown
engines/e_afalg.c:800:AFALG unknown
engines/e_capi.c:290:CAPI unknown
engines/e_capi.c:296:CAPI unknown
engines/e_capi.c:323:CAPI unknown
engines/e_capi.c:350:CAPI unknown
engines/e_capi.c:370:CAPI unknown
engines/e_capi.c:386:CAPI unknown
engines/e_capi.c:489:CAPI unknown
engines/e_capi.c:637:CAPI unknown
engines/e_capi.c:648:CAPI unknown
engines/e_capi.c:655:CAPI unknown
engines/e_capi.c:667:CAPI unknown
engines/e_capi.c:713:CAPI unknown
engines/e_capi.c:760:CAPI unknown
engines/e_capi.c:778:CAPI unknown
engines/e_capi.c:792:CAPI unknown
engines/e_capi.c:814:CAPI unknown
engines/e_capi.c:835:CAPI unknown
engines/e_capi.c:867:CAPI unknown
engines/e_capi.c:875:CAPI unknown
engines/e_capi.c:882:CAPI unknown
engines/e_capi.c:890:CAPI unknown
engines/e_capi.c:931:CAPI unknown
engines/e_capi.c:938:CAPI unknown
engines/e_capi.c:945:CAPI unknown
engines/e_capi.c:954:CAPI unknown
engines/e_capi.c:995:CAPI unknown
engines/e_capi.c:1000:CAPI unknown
engines/e_capi.c:1006:CAPI unknown
engines/e_capi.c:1013:CAPI unknown
engines/e_capi.c:1021:CAPI unknown
engines/e_capi.c:1065:CAPI unknown
engines/e_capi.c:1102:CAPI unknown
engines/e_capi.c:1107:CAPI unknown
engines/e_capi.c:1112:CAPI unknown
engines/e_capi.c:1128:CAPI unknown
engines/e_capi.c:1134:CAPI unknown
engines/e_capi.c:1142:CAPI unknown
engines/e_capi.c:1197:CAPI unknown
engines/e_capi.c:1204:CAPI unknown
engines/e_capi.c:1211:CAPI unknown
engines/e_capi.c:1221:CAPI unknown
engines/e_capi.c:1238:CAPI unknown
engines/e_capi.c:1272:CAPI unknown
engines/e_capi.c:1277:CAPI unknown
engines/e_capi.c:1327:CAPI unknown
engines/e_capi.c:1389:CAPI unknown
engines/e_capi.c:1496:CAPI unknown
engines/e_capi.c:1501:CAPI unknown
engines/e_capi.c:1590:CAPI unknown
engines/e_capi.c:1632:CAPI unknown
engines/e_capi.c:1641:CAPI unknown
engines/e_capi.c:1824:CAPI unknown
engines/e_capi.c:1835:CAPI unknown
engines/e_dasync.c:195:DASYNC unknown
engines/e_dasync.c:210:DASYNC unknown
engines/e_dasync.c:641:DASYNC unknown
engines/e_ossltest.c:318:OSSLTEST unknown
crypto/srp/srp_vfy.c:192:SRP unknown
````

**Comment by levitte** [comment]
I discovered that long list of errors a few moments ago. Will fix in a new PR, where I will teach the script to take the same config file option as mkerr.pl.

**Comment by levitte** [comment]
(also, it seems our Travis script is flawed, or this fault would have been discovered before merging)

**Comment by DDvO** [comment]
These inconsistencies are likely not the only ones, in particular when new sub-libraries/modules are added, given the rather intricate way they are integrated,in the crypto lib with the various meta files that need to be kept in sync. Well, this is supported by various Perl scripts, yet they have their complexity in themselves and non-trivial interactions also with the Makefile. 

Would be nice if sub-library/module integration could be defined and managed more centrally, minimizing redundancies.

**Comment by richsalz** [comment]
> Would be nice if sub-library/module integration could be defined and managed more centrally, minimizing redundancies.

YES!  Can you open an issue for this?  Let's try to address it in one place with one file.

**Comment by DDvO** [comment]
Done.

**Pull request closed without merging**